packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -348,8 +346,18 @@ class TreeViewExtImpl<T> implements Disposable {
         return idLabel;
     }
 
-    private buildTreeItemId(parentId: string, index: number, idLabel: string | undefined): string {
-        return `${parentId}/${index}:${idLabel}`;
+    private buildTreeItemId(parentId: string, item: TreeItem): string {
+        // build tree id according to https://code.visualstudio.com/api/references/vscode-api#TreeItem
```

Comment:
Do those docs prescribe or describe any particular form for the ID or any algorithm for its derivation? I only see that they ID should be a string.

Reply:
Yes, if you look at the doc for the id field. 

>  Optional id for the tree item that has to be unique across tree. The id is used to preserve the selection and expansion state of the tree item.
If not provided, an id is generated using the tree item's label. Note that when labels change, ids will change and that selection and expansion state cannot be kept stable anymore.

Reply:
I see it now - the color they use for the expansion toggles was light enough that I believed that it was a 'disabled' color.